packages/types/src/primitive/Method.ts

```diff
@@ -127,14 +128,21 @@ export default class Method extends Struct implements IMethod {
     return {
       args: new Uint8Array(),
       argsDef: {},
-      meta: new FunctionMetadata(),
+      meta: new MetaV0(),
       callIndex: new Uint8Array([255, 255])
     };
   }
 
   // If the extrinsic function has an argument of type `Origin`, we ignore it
-  static filterOrigin (meta?: FunctionMetadata): Array<FunctionArgumentMetadata> {
+  static filterOrigin (meta?: MetaV0 | MetaV4): Array<FunctionArgumentMetadata> {
     // FIXME should be `arg.type !== Origin`, but doesn't work...
+    if (meta instanceof MetaV4) {
+      return meta
+      ? meta.args.filter(({ type }) =>
```

Comment:
Not for the PR, since it is the way it is - but we don't receive `Origin` anymore, so something to keep in mind to remove. (@amaurymartiny possibly when you rework?)